codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -1026,7 +1057,11 @@ void readEOS() {
 
             void notifyReadComplete(Handle allocHandle) {
                 assert next == null && previous == null;
-                readInProgress = false;
+                if (readStatus == ReadStatus.REQUESTED) {
+                    readStatus = ReadStatus.IN_PROGRESS;
```

Comment:
Are you sure this is all you have to do? If auto-read is disabled on the parent channel, might you not have to request a read to ensure progress? (This may happen somewhere else, but I can't see where.)

Reply:
@rschmitt that is a different "issue" and I am not even sure if we should do anything here at all by default. You could easily add a `ChannelOutboundHandler` to the `ChannelPipeline` of the Channel which will trigger `channel.parent().read()` when a read is received.

Reply:
We could also make this a `ChannelOption` on the "child channel".

Reply:
Just to give some more background why I think we should not handle this by default... I think the user may want to handle the read pattern / back pressure independently in some cases.

Reply:
If this is the philosophy behind this class, then I have no idea why my current h2 server works, because it only reads from the parent channel one time (right at the end of `initChannel`), and autoread is constantly disabled on both parent and child streams.

Reply:
I will investigate once this one is pulled in... Like I said I think these are different "concerns".

Reply:
@rschmitt now that this one is merged I will check your other concern during this week.